joomla/joomla-cms
# Block Usage of User Registration emails for phishing/malware spam

Pull Request for Issue # .

### Summary of Changes

Add new check to com_users User Table check on data validity to stop Joomla from being used to distribute phishing/malware emails.

### Testing Instructions

In unaltered Joomla site make sure that user registrations are enabled and in the frontend create a new user with data similar to the following (use plain text not HTML tags for the links)

name: Fred Flintstone.  You have been selected to win $1000 - simply download your claim form from https://www.dodgywebsite.com/dubiousfile.html

username:fred.    You have been selected to win $1000 - simply download your claim form from https://www.dodgywebsite.com/dubiousfile.html

Then fill the rest of the form with valid data.

### Expected result

If you do not apply this patch the email address used in the form gets an email that tempts them to click the link in the message body - or your Joomla site could get flagged as a distributor of malware.  Note that some email servers may block the sending of this type of message but most will not.

This blatant attempt to send malware using Joomla should be blocked

### Actual result

Dodgy email is sent!

Apply the patch and the username and name fields are blocked from including http:// or https:// - unfortunately this still doesn't block names/usernames containing valid URLs which many email packages will still render as clickable links :(

The only way I can see to block this is to stop usernames and names from including a full stop/period.  Not sure of the implications of that - probably ok if we only apply the restriction to new accounts 

e.g. 
```php
		if ($this->id !== 0 && (strpos($this->username, '.') > 0 || strpos($this->name, '.') > 0))
		{
			$this->setError(\JText::sprintf('JLIB_DATABASE_ERROR_USERNAME_NAME_MUST_NOT_CONTAIN_PERIOD', 2));

			return false;			
		}
```
### Documentation Changes Required
